### ISSUE TITLE
The error occurs when downloading with the argument "--clip_col='timestamp'".

### ISSUE THREAD

**Issue by peiliu0408** [snap-research/Panda-70M#10]
Before downloading the dataset, an experiment was conducted using 200 videos. While these 200 videos could be downloaded (I got 193 videos finally) correctly without the argument "--clip_col="timestamp".

But an error occurred when this argument was included. In the subsequent experiment, none of the videos were downloaded correctly.

<img width="730" alt="截屏2024-03-07 16 51 31" src="https://github.com/snap-research/Panda-70M/assets/18061950/f1ab9af4-8774-4d43-966c-b63b8aa89e83">
<img width="1153" alt="截屏2024-03-07 16 52 05" src="https://github.com/snap-research/Panda-70M/assets/18061950/c8a86929-19f9-410c-bcd3-184192b76cef">

I am new to this tool, so I am unsure if there might be any formatting errors in the timestamp.

**Comment by pabl0** [comment]
It seems to me you are using the upstream video2dataset tool instead of the one included in this repository, which does not work with the provided .csv files that includes the time stamps as Python arrays which are not parsed properly by the normal video2dataset, and gets passed as a string in `clip_spans`.

If you want to use the standard video2dataset tool, you might want to convert the csv files to proper JSON. And it would be nice if this repo documented the changes to "vendored" copy of video2dataset and made it clear that the normal command does not work. Maybe also consider publishing the metadata in better format than csv.

**Comment by tsaishien-chen** [comment]
@peiliu0408 @pabl0 
Yes, it seems you are using video2dataset for downloading, but it cannot work on Panda70M csv files.
Please check https://github.com/snap-research/Panda-70M/issues/3#issuecomment-1984201790 for the reason to modify video2dataset tool
And please use the video2dataset in this repo to download the dataset.

**Comment by tsaishien-chen** [comment]
@peiliu0408 @pabl0 
Yes, it seems you are using video2dataset for downloading, but it cannot work on Panda70M csv files.
Please check [here](https://github.com/snap-research/Panda-70M/issues/3#issuecomment-1984201790) for the reason to modify video2dataset tool.
And please try to use the video2dataset in this repo to download the dataset.

**Comment by peiliu0408** [comment]
> It seems to me you are using the upstream video2dataset tool instead of the one included in this repository, which does not work with the provided .csv files that includes the time stamps as Python arrays which are not parsed properly by the normal video2dataset, and gets passed as a string in `clip_spans`.
> 
> If you want to use the standard video2dataset tool, you might want to convert the csv files to proper JSON. And it would be nice if this repo documented the changes to "vendored" copy of video2dataset and made it clear that the normal command does not work. Maybe also consider publishing the metadata in better format than csv.


You're welcome! Just to confirm, I installed the official video2dataset version and now I'd like to reinstall the tool from this repository

**Comment by tsaishien-chen** [comment]
Yes, please try to uninstall the original video2dataset and reinstall our version!

**Comment by peiliu0408** [comment]
> Yes, please try to uninstall the original video2dataset and reinstall our version!

I have some questions regarding how to set the optimal processes_count and thread_count in the config file for a machine with 8 cores and 16GB of memory. Can you help me with this?

**Comment by peiliu0408** [comment]
> > Yes, please try to uninstall the original video2dataset and reinstall our version!
> 
> I have some questions regarding how to set the optimal processes_count and thread_count in the config file for a machine with 8 cores and 16GB of memory. Can you help me with this?

I also did not find any videos in the download output folder, similar to the picture shown below.
![Uploading 截屏2024-03-08 11.12.44.png…]()
